packages/contracts-bedrock/test/invariants/OptimismSuperchainERC20.t.sol

```diff
@@ -175,10 +191,35 @@ contract OptimismSuperchainERC20_SendSucceeds_Invariant is Test {
     ///                   Actor's balance should also not increase out of nowhere.
     function invariant_sendERC20_succeeds() public view {
         // Assert that the actor has not failed to send OptimismSuperchainERC20.
-        assertEq(actor.failed(), false);
+        assertTrue(!actor.failed(), "0");
+
+        // Assert that the actor has sent more than or equal to the amount relayed.
+        assertTrue(actor.totalAmountSent() >= actor.totalAmountRelayed());
+
+        // Assert that the actor's balance has decreased by the amount sent.
+        assertEq(optimismSuperchainERC20.balanceOf(address(actor)), STARTING_BALANCE - actor.totalAmountSent());
+
+        // Assert that the total supply of the OptimismSuperchainERC20 contract has decreased by the amount unrelayed.
+        uint256 _unrelayedAmount = actor.totalAmountSent() - actor.totalAmountRelayed();
+        assertEq(optimismSuperchainERC20.totalSupply(), STARTING_BALANCE - _unrelayedAmount);
+    }
+
+    /// @notice Invariant that checks that relaying OptimismSuperchainERC20 always succeeds.
+    /// @custom:invariant Calls to relayERC20 should when a message is received from another chain.
+    ///                   Actor's balance should get his amount minted when
+    /// the message is realyed and the amount is greater than 0.
+    function invariant_relayERC20_succeeds() public view {
```

Comment:
Arent we testing the same as `invariant_sendERC20_succeeds` ? With only 1 assert being different

Reply:
Not exactly, there are 2 assertions being different, and to me is cleaner to have them separately since those are different flows